internal/core/src/segcore/InsertRecord.h

```diff
@@ -34,8 +35,47 @@ struct InsertRecord {
     // used for timestamps index of sealed segment
     TimestampIndex timestamp_index_;
 
+    // pks to row offset
+    Pk2OffsetType pk2offset_;
+
     explicit InsertRecord(const Schema& schema, int64_t size_per_chunk);
 
+    std::vector<SegOffset>
+    search_pk(const PkType pk, Timestamp timestamp) const {
+        std::vector<SegOffset> res_offsets;
+        auto [iter_b, iter_e] = pk2offset_.equal_range(pk);
+        for (auto iter = iter_b; iter != iter_e; ++iter) {
+            auto offset = SegOffset(iter->second);
+            if (timestamps_[offset.get()] <= timestamp) {
+                res_offsets.push_back(offset);
+            }
+        }
+
+        return std::move(res_offsets);
```

Comment:
no need to call std::move, it prevents RVO.

Reply:
fixed